odbc_fdw.c

```diff
@@ -2162,6 +2162,24 @@ odbcImportForeignSchema(ImportForeignSchemaStmt *stmt, Oid serverOid)
 					}
 					appendStringInfo(&col_str, "\"%s\" %s", ColumnName, (char *) sql_type.data);
 				}
+				#ifdef DEBUG
+				if (ret == SQL_ERROR || ret == SQL_SUCCESS_WITH_INFO)
+				{
+					SQLINTEGER   j = 1;
+					SQLINTEGER   native;
+					SQLCHAR  state[ 7 ];
+					SQLCHAR  text[256];
+					SQLSMALLINT  len;
+					SQLRETURN    diag_ret;
+					do
+					{
+				        diag_ret = SQLGetDiagRec(SQL_HANDLE_STMT, columns_stmt, j++, state, &native, text, sizeof(text), &len);
+						if (SQL_SUCCEEDED(diag_ret))
+							elog(DEBUG1, "FETCHING %s:%ld:%ld:%s\n", state, (long int) j, (long int) native, text);
+					}
+					while( diag_ret == SQL_SUCCESS );
```

Comment:
from my ignorance: don't you want to log anything when `ret == SQL_ERROR` ? or is that covered by the `diag_ret` ?

Reply:
Note that `diag_ret` is the result of calling [`GetDescRec`](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlgetdescrec-function) to get diagnostic messages iteratively (there may be multiple messages). According to documentation `SQL_NO_DATA` should signal the end of the iteration, but I'm stopping at anything other than `SQL_SUCCESS` to avoid handling errors here 🙈 (hopefully unlikely)